packages/flutter_goldens/lib/src/flutter_goldens_io.dart

```diff
@@ -0,0 +1,646 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async' show FutureOr;
+import 'dart:convert' show jsonEncode;
+import 'dart:io' as io show File, OSError, SocketException, stdout;
+
+import 'package:file/file.dart';
+import 'package:file/local.dart';
+import 'package:flutter/foundation.dart';
+import 'package:flutter_goldens_client/skia_client.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:platform/platform.dart';
+
+import 'flaky_goldens.dart';
+
+export 'package:flutter_goldens_client/skia_client.dart';
+
+// If you are here trying to figure out how to use golden files in the Flutter
+// repo itself, consider reading this wiki page:
+// https://github.com/flutter/flutter/wiki/Writing-a-golden-file-test-for-package%3Aflutter
+
+const String _kFlutterRootKey = 'FLUTTER_ROOT';
+
+/// {@template flutter.goldens.expectFlakyGolden}
+/// Similar to [matchesGoldenFile] but specialized for Flutter's own tests when
+/// they are flaky.
+///
+/// Asserts that a [Finder], [Future<ui.Image>], or [ui.Image] - the [key] -
+/// matches the golden image file identified by [goldenFile].
+///
+/// For the case of a [Finder], the [Finder] must match exactly one widget and
+/// the rendered image of the first [RepaintBoundary] ancestor of the widget is
+/// treated as the image for the widget. As such, you may choose to wrap a test
+/// widget in a [RepaintBoundary] to specify a particular focus for the test.
+///
+/// The [goldenFile] may be either a [Uri] or a [String] representation of a URL.
+///
+/// Flaky golden file tests are always uploaded to Skia Gold for manual
+/// inspection. This allows contributors to validate when a test is no longer
+/// flaky by visiting https://flutter-gold.skia.org/list,
+/// and clicking on the respective golden test name. The UI will show the
+/// history of generated goldens over time. Each unique golden gets a unique
+/// color. If the color is the same for all commits in the recent history, the
+/// golden is likely no longer flaky and the standard [matchesGoldenFile] can be
+/// used in the given test. If the color changes from commit to commit then it
+/// is still flaky.
+/// {@endtemplate}
+Future<void> expectFlakyGolden(Object key, String goldenFile) {
+  assert(
+    goldenFileComparator is FlutterGoldenFileComparator,
+    'matchesFlutterGolden can only be used with FlutterGoldenFileComparator '
+    'but found ${goldenFileComparator.runtimeType}.'
+  );
+
+  (goldenFileComparator as FlutterGoldenFileComparator).enableFlakyMode();
```

Comment:
I thought the same thing! I wondered how important the assertion is... but if it is all contained in the same mixin, maybe we don't have to assert that the right comparator class is being used? Since it is all mixed in? :)

Reply:
Since this is limited to our own repo, I think we can figure it out quickly if someone attempts to use `expectFlakyGolden` without a `flutter_test_config.dart`.

Reply:
By which I mean: "I agree, the assert isn't that valuable".